share/spice/airlines/airlines.js

```diff
@@ -1,115 +1,255 @@
-            "&flightNumber=" +
-            flight[0].FlightNumber;
+        if (results.length === 0) {
+            return Spice.failed('airlines');
+        }
+
+        // Compute the difference between now and the time of departure or arrival.
+        function relativeTime(date, airportOffset) {
+            // This is the time of departure or arrival (not sure why we're getting the difference).
+            date = date.getTime() - (date.getTimezoneOffset() * MILLIS_PER_MIN);
+
+            // This is the current time at the airport (in milliseconds).
+            var now = new Date().getTime() + (airportOffset * MILLIS_PER_HOUR);
+
+            return date - now;
+        }
+
+        // Check if the airplane is on-time or delayed.
+        function onTime(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival) {
+
+            var deltaDepart = new Date(departureDate) - scheduledDeparture,
+                deltaArrive = new Date(arrivalDate) - scheduledArrival;
+
+            if (flight.status === "A" || flight.status === "S") {
+                if (MILLIS_PER_MIN * 5 < deltaDepart && MILLIS_PER_MIN * 5 < deltaArrive) {
+                    return ["Delayed", false];
+                } else {
+                    return ["On Time", true];
+                }
+            }
+            return [STATUS[flight.status], true];
+        }
+
+        // Add the date and time or departure or arrival.
+        Spice.registerHelper("airline_time", function(isDeparture, arrivalDate, departureDate) {
+            var dateObject = new Date(arrivalDate);
+            if(isDeparture) {
+                dateObject = new Date(departureDate);
+            }
+
+            var hours = dateObject.getHours(),
+                minutes = dateObject.getMinutes(),
+                date = dateObject.toDateString();
+
+            date = date.substring(0, date.lastIndexOf(" "));
+
+            // AM or PM?
+            var suffix = (hours >= 12) ? "PM" : "AM";
+
+            // Convert to 12-hour time.
+            if(hours > 12) {
+                hours -= 12;
+            } else if(hours === 0) {
+                hours = 12;
+            }
+
+            // Add leading zeroes.
+            minutes = minutes < 10 ? "0" + minutes : minutes;
+
+            if ((arrivalDate.getDate() != departureDate.getDate()) && (!isDeparture)) {
+                return hours + ":" + minutes + " " + suffix + " (" + MONTH[dateObject.getMonth()] + " " + dateObject.getDate() + ")";
+            } else
+                return hours + ":" + minutes + " " + suffix;
+        });
+
+        Spice.registerHelper("airline_status", function(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival) {
+            var result = onTime(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival),
+                ok_class = result[1] ? "tile__ok" : "tile__not";
+            return '<div class="' + ok_class + '">' + result[0] + '</div>';
+        });
```

Comment:
Nitpick: I'd like `relativeTime`, `onTime`, `airline_time`, and `airline_status` out of `ddg_spice_airlines` if possible just to be consistent with the other IAs.